### ISSUE TITLE
[BUG]: Object allocator is not copy asignable

### ISSUE THREAD

**Issue by rianquinn** [Bareflank/hypervisor#689]
At the moment, the object allocator attempts to adhere to the allocator concept but fails to adhere properly as it is currently not copy assignable. To fix this, some work will need to be done to allow the allocator to be copied, which will also require a mutex to implement properly.

**Comment by rianquinn** [comment]
Ok, so the issue with the object allocator is that:
- If an allocator is copied, the copy and the original _must_ be equal. Since allocator must also support rebinding (i.e. changing the allocator's type), this means that a copy of an allocator (even if the types are different) must be able to allocate/deallocate each other's memory. 
- Our object allocator is a pool allocator meant for allocating and deallocating memory for different sizes (similar to the SLAB for Linux). It is not able to allocate a type of one size, and deallocate another (the type must always be the same). 

With all of this in mind, and the fact that nobody is using the allocator interface, and the fact that the SLAB allocator used by malloc(), free() largely deals with external fragmentation, there is no need for this interface. There is certainly no reason to overcomplicate the implementation of our object allocator just for the sake of adhering to the broken allocator concept. 

Long story short, we will be removing this interface.

**Comment by connojd** [comment]
:+1:

**Comment by rianquinn** [comment]
Going to close this as it has been addressed in the hyperkernel PR that will be merged in a couple of weeks.
